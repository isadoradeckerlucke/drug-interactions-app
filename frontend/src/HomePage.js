import React from "react";
import { useSelector } from "react-redux";
import LoggedInHome from "./LoggedInHome";

function HomePage() {
  const currentUser = useSelector((st) => st.currentUser);
  // for users who are logged in but have no drugs saved
  if (
    currentUser.isLoggedIn &&
    currentUser.currentUser.savedDrugs.length === 0
  ) {
    return (
      <div>
        <h1>
          Welcome to the Drug Interaction Hub,{" "}
          {currentUser.currentUser.username}
        </h1>
        <p>you haven't saved any medications yet!</p>
      </div>
    );
  }

  return (
    <div>
      {currentUser.isLoggedIn ? (
        <LoggedInHome currentUser={currentUser} />
      ) : (
        <div>
          <h2>Welcome to the Drug Interaction Hub</h2>
          <h4>log in or sign up to see your saved medications</h4>
        </div>
      )}
      <h6 muted>
        disclaimer: this site can not provide specific medical advice, and is
        only intended to provide more information to users. please consult with
        a physician before making medical decisions.
      </h6>
    </div>
  );
}

export default HomePage;
