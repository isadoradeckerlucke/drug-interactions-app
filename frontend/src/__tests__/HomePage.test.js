import React from "react";
import { render } from "@testing-library/react";
import HomePage from "../HomePage";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router";
import configureStore from "redux-mock-store";

describe("home page works as expected", function () {
  const initialState = {
    searchTerm: { searchTerm: "adderall" },
    currentUser: {
      isLoggedIn: true,
      currentUser: {
        username: "testuser",
        email: "testuser@gmail.com",
        savedDrugs: [],
      },
    },
  };
  const initialStateLoggedOut = {
    searchTerm: { searchTerm: "adderall" },
    currentUser: {
      isLoggedIn: false,
      currentUser: {},
    },
  };
  const mockStore = configureStore();
  let store;

  /** smoke test */
  it("renders without crashing, logged in user", function () {
    store = mockStore(initialState);
    render(
      <MemoryRouter>
        <Provider store={store}>
          <HomePage />
        </Provider>
      </MemoryRouter>
    );
  });

  /** snapshot test */
  it("matches snapshot, logged in user", function () {
    store = mockStore(initialState);
    const { asFragment } = render(
      <MemoryRouter>
        <Provider store={store}>
          <HomePage />
        </Provider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  /** smoke test */
  it("renders without crashing, logged out user", function () {
    store = mockStore(initialStateLoggedOut);
    render(
      <MemoryRouter>
        <Provider store={store}>
          <HomePage />
        </Provider>
      </MemoryRouter>
    );
  });

  /** snapshot test */
  it("matches snapshot, logged out user", function () {
    store = mockStore(initialStateLoggedOut);
    const { asFragment } = render(
      <MemoryRouter>
        <Provider store={store}>
          <HomePage />
        </Provider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
