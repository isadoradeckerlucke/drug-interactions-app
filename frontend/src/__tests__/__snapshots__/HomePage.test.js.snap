// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home page works as expected matches snapshot, logged in user 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Welcome to the Drug Interaction Hub, testuser
    </h1>
    <p>
      you haven't saved any medications yet!
    </p>
  </div>
</DocumentFragment>
`;

exports[`home page works as expected matches snapshot, logged out user 1`] = `
<DocumentFragment>
  <div>
    <div>
      <h2>
        Welcome to the Drug Interaction Hub
      </h2>
      <h4>
        log in or sign up to see your saved medications
      </h4>
    </div>
    <h6>
      disclaimer: this site can not provide specific medical advice, and is only intended to provide more information to users. please consult with a physician before making medical decisions.
    </h6>
  </div>
</DocumentFragment>
`;
